ci-operator/config/openshift/hive/master.yaml

```diff
@@ -0,0 +1,61 @@
+base_images:
+  base:
+    cluster: https://api.ci.openshift.org
+    name: origin-v4.0
+    namespace: openshift
+    tag: base
+  root:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+binary_build_commands: make build
+canonical_go_repository: github.com/openshift/hive
+images:
+- dockerfile_path: build/hive/Dockerfile
+  from: base
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/openshift/hive/bin/manager
+  to: hive
+- dockerfile_path: build/build-image/Dockerfile
+  from: root
+  optional: true
+  to: builder-base
+- dockerfile_path: build/test-image/Dockerfile
+  from: builder-base
+  optional: true
+  to: test-base
+resources:
+  '*':
+    limits:
+      cpu: '2'
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+  unit:
+    limits:
+      cpu: '4'
+      memory: 4Gi
+    requests:
+      cpu: '2'
+      memory: 1Gi
+tag_specification:
+  cluster: https://api.ci.openshift.org
+  name: origin-v4.0
+  namespace: openshift
+  tag: ''
+  tag_overrides: {}
+build_root:
+  image_stream_tag:
```

Comment:
I think what you want is to use `project_image` here, not the tag, and set the project build to use your builder-base.

Anything like:
```
# Create the full dir tree that we'll mount our src into when we run the image
RUN mkdir -p /go/src/github.com/openshift/hive

# Default to our src dir
WORKDIR /go/src/github.com/openshift/hive
```
is an implementation detail of the CI and you should not need to do it at all

you should not need your test-base image